### ISSUE TITLE
[LAB] Week 1_Problem 2_Dongwahn Kim

### ISSUE THREAD

**Issue by dongwahn78** [vlab-kaist/NN101_23S#21]
### Problem

Week 1_Problem 2

### Source Code

```python
import torch
from random import random 
from typing import Callable


def solution(func: Callable, start_point: float) -> float:
    x = torch.tensor(start_point, requires_grad=True)
    y = func(x)
    while True:
        grad, = torch.autograd.grad(y, x, create_graph=True)
        x = x - 0.01 * grad
        y = func(x)
        if grad.abs() < 1e-5:
            break
    return x.item()


if __name__ == "__main__":
    def test_func(x):
        return x ** 4
    t = 10*random()
```

### Description

done

**Comment by jangyoujin0917** [comment]
Please do not use optim in PyTorch.
This problem is for checking your understanding for gradient-based optimization.
